File: src/main/java/it/giaquinto/springberry/model/user/implementation/AdminUser.java
package it.giaquinto.springberry.model.user.implementation;

import it.giaquinto.springberry.model.security.SecurityLevel;

import java.time.LocalDateTime;

public final class AdminUser extends BaseUser {

    public AdminUser(
            final String name,
            final String surname,
            final LocalDateTime created
    ) {
        super(name, surname, created);
    }

    @Override
    public SecurityLevel levelOf() {
        return SecurityLevel.ADMIN;
    }


}

File: src/main/java/it/giaquinto/springberry/model/user/implementation/AdminUser.kt
package it.giaquinto.springberry.model.user.implementation

import it.giaquinto.springberry.model.security.SecurityLevel
import java.time.LocalDateTime

class AdminUser(
    name: String?,
    surname: String?,
    created: LocalDateTime?
) : BaseUser(name, surname, created) {
    override fun levelOf(): SecurityLevel? {
        return SecurityLevel.ADMIN
    }
}